pkgs/native_toolchain_c/test/cbuilder/cbuilder_test.dart

```diff
@@ -200,6 +200,222 @@ void main() {
       () => testDefines(customDefineWithValue: value),
     );
   }
+
+  test('CBuilder flags', () async {
+    await inTempDir(
+      // https://github.com/dart-lang/sdk/issues/40159
```

Comment:
We do have a `close` now.

Reply:
Yeah, on Windows we can't delete files when they are in use. And when we dlopen a dll, we can't then delete it.
On unix based systems, you can delete files when they are in use.

Anyway, feel free to leave this for another PR.

Reply:
I've refactored `inTempDir` to `tempDirForTest` which uses the `addTearDown`. I quite like to build test helpers that clean up resources like that because they don't have the additional nesting of a function. But please let me know if you would prefer a more explicit clean up.

Reply:
Oh neat! :rocket: